src/codemagic/tools/app_store_connect/actions/submit_to_app_store_action.py

```diff
@@ -0,0 +1,504 @@
+        promotional_text: Optional[str],
+        support_url: Optional[str],
+        whats_new: Optional[Union[str, Types.WhatsNewArgument]] = None,
+    ) -> List[AppStoreVersionLocalizationInfo]:
+        if isinstance(app_store_version_localizations, Types.AppStoreVersionLocalizationInfoArgument):
+            app_store_version_localization_infos = app_store_version_localizations.value
+        else:
+            app_store_version_localization_infos = app_store_version_localizations or []
+
+        app_store_version_localization_info = AppStoreVersionLocalizationInfo(
+            description=description,
+            keywords=keywords,
+            locale=locale,
+            marketing_url=marketing_url,
+            promotional_text=promotional_text,
+            support_url=support_url,
+            whats_new=whats_new.value if isinstance(whats_new, Types.WhatsNewArgument) else whats_new,
+        )
+        if set(app_store_version_localization_info.__dict__.values()) != {None}:
+            # At least some field is defined
+            app_store_version_localization_infos.append(app_store_version_localization_info)
+
+        return app_store_version_localization_infos
+
+    def _manage_app_store_version_phased_release(
+        self,
+        app_store_version: AppStoreVersion,
+        should_enable_phased_release: Optional[bool],
+    ):
+        if should_enable_phased_release is True:
+            self._enable_app_store_version_phased_release(app_store_version)
+        elif should_enable_phased_release is False:
+            self._disable_app_store_version_phased_release(app_store_version)
+        else:
+            pass  # Leave it as is without changing anything
+        self.echo("")
+
+    def _enable_app_store_version_phased_release(self, app_store_version: AppStoreVersion):
+        self.echo(Colors.BLUE(f"\nEnable phased release for App Store Version {app_store_version.id}"))
+        phased_release = self.api_client.app_store_versions.read_app_store_version_phased_release(app_store_version)
+        if phased_release:
+            self.echo(Colors.GREEN(f"Phased release is already enabled for App Store Version {app_store_version.id}"))
+        else:
+            phased_release = self.enable_app_store_version_phased_release(app_store_version, should_print=False)
+        self.printer.print_resource(phased_release, should_print=True)
+
+    def _disable_app_store_version_phased_release(self, app_store_version: AppStoreVersion):
+        self.echo(Colors.BLUE(f"\nDisable phased release for App Store Version {app_store_version.id}"))
+        phased_release = self.api_client.app_store_versions.read_app_store_version_phased_release(app_store_version)
+        if not phased_release:
+            self.echo(Colors.GREEN(f"Phased release is already disabled for App Store Version {app_store_version.id}"))
+        else:
+            self.cancel_app_store_version_phased_release(phased_release)
```

Comment:
Genuinely new code, along with usage of `_manage_app_store_version_phased_release` in `_submit_to_app_store`.